File: web/src/app/equipment/equipment.ts
// A machine as the API describes it. The dates come as ISO strings and
// exist only while the machine is out.
export interface Equipment {
  id: number;
  name: string;
  category: string;
  description: string;
  dailyRate: number;
  imageUrl: string;
  available: boolean;
  rentedFrom: string | null;
  rentedUntil: string | null;
}

export interface Rental {
  id: number;
  equipmentId: number;
  startsOn: string;
  endsOn: string;
  days: number;
  total: number;
}

// The durations a card offers, in days.
export const DURATIONS = [1, 3, 5, 7];

File: api/EquipmentRental.Api/Models/Equipment.cs
namespace EquipmentRental.Api.Models
{
    // A machine in the fleet. The picture is a public photo linked by
    // URL, nothing is stored in the repository.
    public class Equipment
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Category { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public decimal DailyRate { get; set; }
        public string ImageUrl { get; set; } = string.Empty;
        public List<Rental> Rentals { get; set; } = new();
    }
}
